File: scripts/test-api-dns.py
import httpx

# Test DNS resolution of fazle-api from brain
try:
    r = httpx.get("http://fazle-api:8100/health", timeout=5)
    print("fazle-api:8100 ->", r.status_code, r.text)
except Exception as e:
    print("fazle-api:8100 FAIL:", e)

# Test DNS resolution of fazle-api-blue from brain  
try:
    r = httpx.get("http://fazle-api-blue:8100/health", timeout=5)
    print("fazle-api-blue:8100 ->", r.status_code, r.text)
except Exception as e:
    print("fazle-api-blue:8100 FAIL:", e)

File: scripts/test-api-dns.js
const http = require("http");

function testUrl(url) {
  return new Promise((resolve) => {
    http.get(url, (r) => {
      let d = "";
      r.on("data", (c) => d += c);
      r.on("end", () => resolve(url + " -> " + r.statusCode + " " + d));
    }).on("error", (e) => resolve(url + " FAIL: " + e.message));
  });
}

Promise.all([
  testUrl("http://fazle-api:8100/health"),
  testUrl("http://fazle-api-blue:8100/health"),
]).then(results => results.forEach(r => console.log(r)));
